src/hotspot/share/prims/jvmtiEnvBase.cpp

```diff
@@ -1756,6 +1756,7 @@ JvmtiEnvBase::suspend_thread(oop thread_oop, JavaThread* java_thread, bool singl
   HandleMark hm(current);
   Handle thread_h(current, thread_oop);
   bool is_virtual = java_lang_VirtualThread::is_instance(thread_h());
+  bool is_thread_carrying = is_thread_carrying_vthread(java_thread, thread_h());
```

Comment:
I think that somewhere in this place should be an explanation of suspend<->resume synchronization. As I understand the hadshake can't be executed and clear suspend state while suspend_thread is done for the same thread. How it is guaranteed that suspend_thread flag cann't be updated?
It is not obvious and also put some restrictions on the suspend_thread implementation to keep this behaviour.

Reply:
Thank you for reviewing and this suggestion.
Yes, you are right. I'll try to find a good place to add such a comment.

Reply:
I've added a comment you requested. Please, let me know if it is enough or there are some comments/suggestions.